### DISCUSSION TITLE
How to parameterise certain path elements for any discovered endpoint

### DISCUSSION THREAD

**Discussion by kwmlodozeniec** [schemathesis/schemathesis#1449] [Q&A] [answered]
I have a simple API I am trying to test and would like to use fixed data for all the path parameters. I also want to rely on the auto generation of test cases from the Open API spec.

For example, I have this "service"

```
GET /api/service/version
GET /api/service/data/process1/{param1}/{param2}
POST /api/service/trigger/process1/{param1}
POST /api/service/trigger/process2/{param1}
```

How can I specify what values to use for param1 and param2 and still take advantage of the auto discovery of endpoints? I essentially want to have a single test case that processes the whole API based on a spec I load. Below is an example code of how I would like to run this with pytest.

```
import schemathesis

schema = schemathesis.from_path(
    "openapi.json",
    base_url="https://localhost:3000",
    skip_deprecated_operations=True,
)


def my_check(response, case):
    assert response.status_code == 200


@schema.parametrize()
def test_api(case):
    response = case.call()
    case.validate_response(response, checks=(my_check,))
```

**Comment by kwmlodozeniec** [comment, marked as answer]
I have worked around this by checking if there are any path elements in the case and replacing them with fixed values as and when.

**Reply by Stranger6667** [reply]
Oh, I forgot to answer here :( Sorry for that. 
Yep, the solution is the correct one. You could also use [Hypothesis strategies](https://schemathesis.readthedocs.io/en/stable/python.html#using-additional-hypothesis-strategies) if you have multiple possible options to sample from:

```python
@schema.parametrize()
@schema.given(data=st.data())
def test_api(data, case):
    if "param1" in case.path_parameters:
        case.path_parameters["param1"] = data.draw(st.sampled_from(["a", "b", "c"]))
    ...
```

Note that this approach should be used instead of using the `random` module because Hypothesis records all calls to strategies (like `st.sampled_from`) and therefore can reliably reproduce failures.

**Reply by kwmlodozeniec** [reply]
Nice, that seems much clearer now. Thanks!
